modules/apps/transfer/ibc_module.go

```diff
@@ -130,7 +145,12 @@ func (im IBCModule) OnChanOpenAck(
 	if counterpartyVersion != types.Version {
 		return sdkerrors.Wrapf(types.ErrInvalidVersion, "invalid counterparty version: %s, expected %s", counterpartyVersion, types.Version)
 	}
-	return nil
+
+	if im.app == nil {
+		return nil
+	}
+	// call underlying app's OnChanOpenAck callback with the counterparty app version.
+	return im.app.OnChanOpenAck(ctx, portID, channelID, counterpartyVersion)
```

Comment:
What's the reasoning for now adding callbacks for OnChanOpenTry and OnChanOpenConfirm?

Reply:
not much reason beside adding more flexibility, might be useful for the wired app? if not it should return nil anyways?